js/console.ts

```diff
@@ -87,24 +95,24 @@ function stringify(ctx: ConsoleContext, value: any): string {
 
 // Print strings when they are inside of arrays or objects with quotes
 // tslint:disable-next-line:no-any
-function stringifyWithQuotes(ctx: ConsoleContext, value: any): string {
+function stringifyWithQuotes(ctx: ConsoleContext, value: any, level: number, maxLevel: number): string {
   switch (typeof value) {
     case "string":
       return `"${value}"`;
     default:
-      return stringify(ctx, value);
+      return stringify(ctx, value, level, maxLevel);
   }
 }
 
 // tslint:disable-next-line:no-any
-export function stringifyArgs(args: any[]): string {
+export function stringifyArgs(args: any[], options: ConsoleOptions = {}): string {
   const out: string[] = [];
   for (const a of args) {
     if (typeof a === "string") {
       out.push(a);
     } else {
       // tslint:disable-next-line:no-any
-      out.push(stringify(new Set<any>(), a));
+      out.push(stringify(new Set<any>(), a, 0, options.depth || DEFAULT_MAX_DEPTH));
```

Comment:
Sometimes we just want no depth at all. Perhaps you should add a check where if `options.depth` is 0, take it instead of defaulting to 2. In case of `null`, we could either default it to `0` (implying no depth), `2` (default depth), or `Infinity` (implying lack of depth).

Same for line 136.

Reply:
Sounds good, now `null` and `undefined` (if user does not provide this option) defaults to `2` (default depth). Other values including `0` are treated as valid values.

Reply:
this `null` treated as default to 2 is different with node:

in Node this `null` is treated as unlmited depth

```js
> var o = { a: { b: { c: { d: new Date } } } };
undefined
> console.log(o)
{ a: { b: { c: [Object] } } }
> console.dir(o, { depth: null });
{
  a: { b: { c: { d: 2020-08-19T23:55:58.009Z } } }
}
```